File: testdata/fixtures/kotlin-dirty/src/main/kotlin/com/example/service/OrderService.kt
// Service layer: orchestrates the domain. May import domain + stdlib.
package com.example.service

import kotlin.collections.mutableMapOf

import com.example.domain.Order

class OrderService {
    fun placeOrder(total: Int): Order {
        val log = mutableMapOf<Long, Int>()
        val order = Order(total)
        log[order.id] = order.total
        return order
    }
}

File: testdata/fixtures/scala-clean/src/main/scala/com/example/service/OrderService.scala
// Service layer: orchestrates the domain. May import domain + stdlib.
package com.example.service

import scala.collection.mutable.{Map => MutableMap}

import com.example.domain.Order

final class OrderService:
  def placeOrder(total: Int): Order =
    val log = MutableMap.empty[Long, Int]
    val order = Order(total)
    log(order.id) = order.total
    order
